pkg/processor/runtime/python/py/_nuclio_wrapper_as_server.py

```diff
@@ -0,0 +1,290 @@
+            del self.connections[fileno]
+            self._logger.info(f"Connection {fileno} removed from tracking")
+        sock.close()
+
+        # Cancel and remove the associated task
+        if fileno in self.tasks:
+            task = self.tasks.pop(fileno)
+            if not task.done():
+                task.cancel()
+            self._logger.info(f"Task for connection {fileno} cancelled and removed")
+
+    async def start(self, num_requests=None):
+        """Start the server."""
+        # Create server socket
+        server_sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        server_sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        server_sock.bind((self._host, self._port))
+        server_sock.listen(self._max_connections)
+        server_sock.setblocking(False)
+
+        self._logger.info(f"Server started on {self._host}:{self._port}, max_connections: {self._max_connections}")
+
+        # Register server socket with epoll
+        self.selector.register(server_sock, selectors.EVENT_READ)
+        self.connections[server_sock.fileno()] = server_sock
+        try:
+            while True:
+                events = await self.select_events()
+                # Now awaiting the select operation
+                for key, event in events:
+                    sock = key.fileobj
+                    if sock == server_sock:
+                        # Accept new connection
+                        client_sock, addr = server_sock.accept()
+                        self._logger.info(f"Accepted connection from {addr}")
+                        client_sock.setblocking(False)
+                        self.connections[client_sock.fileno()] = client_sock
+                        task = self._loop.create_task(self._process_connection(client_sock))
+                        self.tasks[client_sock.fileno()] = task
+                    else:
+                        self._logger.error(f"Unexpected event: {event}")
+                await asyncio.sleep(0.2)
+        except KeyboardInterrupt:
+            self._logger.info("Shutting down server")
+        finally:
+            self._logger.info("Closing all connections")
+            for sock in self.connections.values():
+                sock.close()
+
+    async def select_events(self, timeout=0.1):
+        # Run the selector.select in the default executor (this is a blocking operation)
+        events = await self._loop.run_in_executor(None, self.selector.select, timeout)
+        return events
+
+    async def initialize(self):
+
+        # call init_context
+        await self._initialize_context()
+
+        # register to the SIGUSR1 and SIGUSR2 signals, used to signal termination/draining respectively
+        self._register_to_signal()
+
+        await self._send_data_on_control_socket({
+            'kind': 'wrapperInitialized',
+            'attributes': {'ready': 'true'}
+        })
+
+    async def _stop_processing(self):
```

Comment:
where is this being called? should it be called on a termination/drain signal?

Reply:
@TomerShor for now it is only called in shutdown(which is called on termination). Draining isn't applicable for async